doc/manual/syntax.md

```diff
@@ -340,6 +341,15 @@ true
 
 > let a = 1 in let b = 2 in a + b
 3
+
+> let rec f = fun n => if n == 0 then n else n + f (n - 1) in f 10
+55
+
+> let rec fib = fun n => if n <= 2 then 1 else fib (n - 1) + fib (n - 2) in fib 9
+34
+
+> let rec repeat = fun n x => if n <= 0 then null else { head = x, tail = repeat (n - 1) x } in repeat 3 "foo"
+{ head = "foo", tail = { head = "foo", tail = { head = "foo", tail = null }}}
```

Comment:
I think using built-in arrays instead of encoding linked lists through records could read better for a neophyte going through the manual.
```suggestion
> let rec repeat = fun n x => if n <= 0 then [] else repeat (n - 1) x @ [x] in repeat 3 "foo"
["foo", "foo", "foo"]
```